### DISCUSSION TITLE
Next steps for Flynt

### DISCUSSION THREAD

**Discussion by aaronmeder** [flyntwp/flynt#458] [General]
Hi community and especially Bleech Flynt team 👋🏻 

We are having a blast using Flynt for almost all of our custom Wordpress themes we've developed at [Soda Studios](https://soda.today) during the last 18 months - so first of all a big thank you for putting together this fine set of tools! 

I've interestingly followed along the development in the [Timo/next](https://github.com/flyntwp/flynt/tree/Timo/next) branch which seems to take some steps with timber2 (which brings PHP8.1 compatibility), a lot of refactoring, possibly removing jQuery as dependency and so on.

I wanted to ask if you might be willing to share some high level next steps you plan to solve with Flynt going forward? Do you have any hints in terms of a rough timeframe in which you hope to release a next version?

Greetings from Zürich 🇨🇭
Aaron

**Comment by timohubois** [comment]
Hi @aaronmeder, nice to read from you! We're still working on the next version of Flynt and we're planning releasing it as soon as possible - but before, there are a few general things were we have to make a decision. 

However, we would love to hear your feedback though:
- Is there anything you miss in the current version of Flynt? 
- What do you expect from future versions of Flynt?
- What feature would you like to find inside the core of Flynt?

We are pleased reading from you 🚀

**Comment by aaronmeder** [comment]
Hey @timohubois 

Great to hear you are working on releasing an update soon. Flynt really has been the best start package we've used and helped us a lot to build great sites with the simple yet very flexible component approach.

**How we usually work with Flynt**
After the 1.4 release we took some time and made our own Starter based on Flynt. The main idea is to have a ready-to-go starter that fits our workflow and tools best. 

Our main modifications are:
- Our own custom CSS Base
- More flexible responsive Typography setup (font sites, weights and line-heights)
- WPML enabled by default: all fields have `'wpml_cf_preferences' => 3` set already
- For all Sections: flexible top and bottom padding options
- ACF's [Local Json](https://www.advancedcustomfields.com/resources/local-json/) enabled by default
- A Modal Component
- A simple Favicon component for drop-in integration of a [Real Favicon Generator](https://realfavicongenerator.net/) set
- Swapped Swiper for [Flickity](https://flickity.metafizzy.co/) - just because we're more used to it
- A slightly customized version of [ACF Image Mapping Hotspots](https://wordpress.org/plugins/acf-image-mapping-hotspots/) that we use in many placed for focus points on responsive images

**What do you expect from future versions of Flynt?**
Hard to tell - there are always so many personal preferences involved I think😬 In general having a familiarly simple, well extendable way to work, with a under-the-hood technically updated basis for a stable future proof way to build Wordpress websites.

- Regular (like twice a year) updates to composer and npm packages
- Generally trying to remove jQuery as a dependency going forward could gain some web vitals points.

I can't really tell yet if we would fully switch back to standard Flynt or update our own starter. Will need to look at the details in the branch a little closer - but the commits I've seen so far seemed all very reasonable 😉


**General Ideas of Bleech**
It would also be interesting to hear from you guys what general ideas behind the new release are and what challenges you've generally had that you are trying to solve with it.

**Reply by timohubois** [reply]
@aaronmeder in relation to WPML and `wpml_cf_preferences`, I created a drop-in for the `inc` folder that can do this:
https://github.com/awesome-flynt/flynt-add-ons/blob/main/inc/wpmlCustomFieldPreferences.php

What do you think about this?

**Reply by aaronmeder** [reply]
That's a very nice solution you came up with @timohubois! It suits Flynt's core style very well: be flexible, but keep things very simple. Will definitely use this solution...much nicer than declaring the preference on all fields manually. 🙌🏻

**Comment by aslauris** [comment]
Hi @timohubois, thank you for your fantastic work contributing to Flynt. I've been following your contributions on the Timo/next branch, which has been packed with great improvements. Are there any plans for a new Flynt release in the near future? Also, do you know if Timo/next is being used in any production projects? Many thanks!

**Comment by timohubois** [comment]
@aslauris we are currently testing the current next branch and used it in some projects.

From my feelings and my current experience I would say you can use the next branch or a tagged alpha version – with reservation; parts will may change and there are some is not documented, yet.

@domtra implemented a nice and modern JavaScript approach that is definitely worth a look.

But we have some decisions to make, changes to implement and minor improvements on the internal roadmap, so that unfortunately, we can not estimate a exact release date.

However, think releasing the new version in the near future is realistic.

Last but not least: We would be very happy about sharing your first experiences when using the current next from its branch or tag ✌️

**Reply by aslauris** [reply]
@timohubois thanks for a quick reply. Actually, the new JavaScript approach is the main reason we'd like to try out the new version :) I'll update you on our experience if we decide to go this route. Thanks again for your contributions.

**Reply by timohubois** [reply]
@aslauris the new version is released now. Would really happy to hear your thoughts about it ✌️

**Reply by aslauris** [reply]
Hi @timohubois I was really excited when I saw the announcement. Thanks for all the hard work you've done improving Flynt! We're about to start a new project and we'll defo use Flynt 2.0 for it. Actually, we've built a project based on a pre-release version of the `next` branch - so far so good! 
We're developing our own components - loads of them to be honest and the only thing I was really missing is `flynt-generator` compatible with Flynt 2.0. I've started creating one based on [plopjs](https://plopjs.com/) (as a part of Flynt theme). I'll share it with you by creating a pull request - maybe someone will find it useful.

**Comment by aaronmeder** [comment]
Hey @timohubois, Steffen, Dominik and whole Bleech team.
Thanks so much for creating v2 and all the polishing that went in the last weeks 🙌🏻 ❤️ 
Very happy to see it released now that I am back from my summer vacation and start using it with the next projects. Maybe it's only my imagination but the final v2 seems to compile super quickly, love it!
